### ISSUE TITLE
Run as non-root user

### ISSUE THREAD

**Issue by SunilMohanAdapa** [freedombox/Plinth#38]
> Probably shouldn't even run as *www-data*, but as *plinth* user, to compartmentalize permissions as much as possible.

**Comment by jvalleroy** [comment]
To correct what I said in the progress call:
a2query doesn't actually need root, but it's installed in /usr/sbin. So we will just need to use the full path when checking if it's installed.

**Comment by nikhilrayaprolu** [comment]
@SunilMohanAdapa @jvalleroy what would be a better way to identify that we dont have root privileges to run?
http://stackoverflow.com/questions/2806897/what-is-the-best-practices-for-checking-if-the-user-of-a-python-script-has-root

I find that we could use `os.geteuid() != 0` but the above discussion says its not a standard way to do it.So shall we go writing a function for this code and import it where ever necessary?
```
try:
    os.rename('/etc/foo', '/etc/bar')
except IOError as e:
    if (e[0] == errno.EPERM):
       print >> sys.stderr, "You need root permissions to do this, laterz!"
       sys.exit(1)
``` 
or is there any other way we could do it?

**Comment by nikhilrayaprolu** [comment]
secondly let us assume that we have started our application in non-root mode then how about some actions which require root permissions like installation of applications, etc?One way we could do this is by asking for sudo permissions when ever required(basically like the way software updaters do:asking for root permissions whenever required)

**Comment by nikhilrayaprolu** [comment]
tried calling this command  `output = actions.superuser_run('tor', ['get-status'])` https://github.com/freedombox/Plinth/blob/master/plinth/modules/tor/utils.py#L49
without superuser permissions during plinth startup and got this error 
```
[2016-07-26 01:51:36,803] plinth.actions INFO     Executing command - ['/home/nikhil/proj/Plinthfinal/Plinth/actions/tor', 'get-status']
Failed to get unit file state for tor.service: No such file or directory

```

**Comment by jvalleroy** [comment]
@nikhilrayaprolu Here's my understanding of the issue:

Currently plinth runs as the root user. We would like to instead run as a non-root user (named "plinth"). This user will have sudo abilities though. But we want to limit where sudo abilities can be used:
- Only commands run by actions.superuser_run can use sudo.
- All other plinth modules or command cannot use sudo.

I think if we follow this (like a contract) then we don't need to check for root privileges, because we know they are only available when using actions.superuser_run.

**Comment by harry-7** [comment]
@jvalleroy  By plinth user you mean user in the system right. So during install we create a user with sudo permissions and we run plinth from the plinth user. Is that what you meant ?

**Comment by jvalleroy** [comment]
@harry-7 Yes, the plinth user is created by the plinth package's postinst script:
https://sources.debian.net/src/plinth/0.9.4-2/debian/postinst/

**Comment by harry-7** [comment]
@jvalleroy So is  this issue  about running in Operating systems like Debian and Ubuntu etc ?

**Comment by nikhilrayaprolu** [comment]
I am planning something like using os.setuid(<ID of plinth>) at the beginning of plinth/__main__.py .and execute "sudo plinth".this way we could execute the whole script as plinth and when ever sudo permissions are needed that can be taken by subprocesses.does this sound good. or can you please me a better one.

I have tried adding USER and GROUP rows in systemd unit file ( https://github.com/freedombox/Plinth/blob/master/data/lib/systemd/system/plinth.service ) and tried starting it "sudo systemctl start plinth".But my plinth module hasn't started (could not access it at localhost:8000/plinth) ,am I in a wrong way? should I do something more to start plinth in this manner?

**Comment by jvalleroy** [comment]
Fixed in #533.
